docs/DeveloperGuide.md

```diff
@@ -708,113 +754,87 @@ testers are expected to do more *exploratory* testing.
-       Error details shown in the status message.
-          Status bar remains the same.
-
-   11. Test case: `add n/Johnny Doe p/98765431 e/johnd@nus.com rm/06-73
-                   dr/13/09/22 - 15/09/23 ng/1 rq/Kill the insect `<br>
-       Expected: No guest is added, because the is room clean is invalid. Error details shown in the status message.
-       Status bar remains the same.
-
+       Expected: No guest is added, because the **`number of guest` is invalid** (>4).
+       Error details shown in the result display.
 
    12. Let INVALID_REQUEST be a string of 501 characters long. <br>
        Test case: `add n/Johnny Doe p/98765431 e/johnd@nus.com rm/06-73
        dr/13/09/22 - 15/09/23 ng/1 rq/INVALID_REQUEST `<br>
-       Expected: No guest is added, because the request is invalid (>500 characters).
-       Status bar remains the same.
+       Expected: No guest is added, because the **`request` is invalid** (>500 characters).
 
 ### Editing a guest
 
-1. Editing a guest
-
-   1. Prerequisite: Only 1 guest to be edited. The guest's index should exist.
+Prerequisite: Only 1 guest to be edited. The guest's index should exist.
    The guest should exist in GuestBook. The format and content of the command should be valid.
 
-   2. Test case: `edit 1 n/Johnny`<br>
-      Expected: Guest edit successfully, the first guest's name will change from "John Doe" to "Johnny"
+   1. Test case: `edit 1 n/Johnny`<br>
+      Expected: Guest edit successfully, the first guest's `name` will change from "John Doe" to "Johnny"
 
    3. Test case: `edit 99999 n/Johnny`<br>
-      Expected: No guest is edited, because the input index does not exist. Error details shown in the status message.
+      Expected: No guest is edited, because the input **index does not exist**. Error details shown in the result display.
+
+   5. Other incorrect edit commands to try: `edit`, `edit x`, `edit 1 rc/hi`, `edit ...`, `edit 1 n/Jo@`<br>
+      Expected: No guest is edited, because the **parameters are invalid**. Error details shown in the result display.
+### Saving data
 
-   4. Test case: `edit 1 n/Jo@`<br>
-      Expected: No guest is edited, because the name is invalid. Error details shown in the status message.
-      Status bar remains the same.
+1. Dealing with corrupted data files
 
-   5. Other incorrect edit commands to try: `edit`, `edit x`, `edit 1 rc/hi`, `edit ...`<br>
-      Expected: No guest is edited, because the parameters are invalid. Error details shown in the status message.
-      Status bar remains the same.
+   1. A possible cause of a corrupted file could be because of duplicate `name` and/or `room` in the data of the Guest.
+   2. While GuestBook throws an exception when a user tries to add a guest with duplicate `name` and/or `room`, a developer
+   can circumvent this by changing the data in the `guestbook.json` file directly. 
+   3. Similarly, to fix this corrupted file you could either:
```

Comment:
Should this be the user as well?

Reply:
This should remain as "you" as we are referring to the developer and not the hotel managers (the user).